ott/core/initializers.py

```diff
@@ -0,0 +1,305 @@
+  init_dual_b = jnp.zeros_like(b) if lse_mode else jnp.ones_like(b)
+  return init_dual_b
+
+
+def remove_weight_potential(
+    weights: jnp.ndarray, init_dual: jnp.ndarray, lse_mode: bool
+) -> Tuple[jnp.ndarray]:
+  """Cancel dual variables for zero weights.
+
+  Args:
+    weights: array of probability masses
+    init_dual: dual potential array
+    lse_mode: Return log potentials if true. Defaults to True.
+
+  Returns:
+    potential
+  """
+  return jnp.where(weights > 0, init_dual, -jnp.inf if lse_mode else 0.0)
+
+
+def remove_weight_potentials(
+    weights_a: jnp.ndarray, weights_b: jnp.ndarray, init_dual_a: jnp.ndarray,
+    init_dual_b: jnp.ndarray, lse_mode: bool
+) -> Tuple[jnp.ndarray]:
+  """Cancel dual variables for zero weights.
+
+  Args:
+    weights_a: array of probability masses, array of size n
+    weights_b: array of probability masses, array of size m
+    init_dual_a: potential f, array of size n
+    init_dual_b: potential g, array of size m
+    lse_mode: Return log potentials if true. Defaults to True.
+
+  Returns:
+    potentials (f,g)
+  """
+  init_dual_a = remove_weight_potential(weights_a, init_dual_a, lse_mode)
+  init_dual_b = remove_weight_potential(weights_b, init_dual_b, lse_mode)
+  return init_dual_a, init_dual_b
+
+
+class SinkhornInitializer:
+  """Initialzation for Sinkhorn potential f.
+
+  Args:
+    ot_problem: OT problem between discrete distributions of size n and m.
+    lse_mode: Return log potential. Defaults to True.
+
+  Returns:
+    dual potential, array of size n
+  """
+
+  def init_dual_a(
+      self, ot_problem: linear_problems.LinearProblem, lse_mode: bool
+  ) -> jnp.ndarray:
+
+    return default_dual_a(ot_problem=ot_problem, lse_mode=lse_mode)
+
+  def init_dual_b(
+      self, ot_problem: linear_problems.LinearProblem, lse_mode: bool
+  ) -> jnp.ndarray:
+    """Initialzation for Sinkhorn potential g.
+
+    Args:
+      ot_problem: OT problem between discrete distributions of size n and m.
+      lse_mode: Return log potential. Defaults to True.
+
+    Returns:
+      dual potential, array of size m
+    """
+    return default_dual_b(ot_problem=ot_problem, lse_mode=lse_mode)
+
+
+class GaussianInitializer(SinkhornInitializer):
+  """GaussianInitializer.
+
+  Args:
+    stop_gradient: Defaults to True.
+  """
+
+  def __init__(self, stop_gradient: bool = True) -> None:
+
+    super().__init__()
+
+    self.stop_gradient = stop_gradient
+
+  def init_dual_a(
+      self,
+      ot_problem: linear_problems.LinearProblem,
+      lse_mode: bool,
+  ) -> jnp.ndarray:
+    """Gaussian init function.
+
+    Args:
+      ot_problem: OT problem description with geometry and weights.
+      init_f: Pre dual sort initialization, when none sets entries as 0.
+      lse_mode: Return log potential if true. Defaults to True.
+
+    Returns:
+      potential f, array of size n.
+    """
+    # import here due to circular imports
+    from ott.tools.gaussian_mixture.gaussian import Gaussian
```

Comment:
remark about consistency of imports

Reply:
Changed, even gaussian.Gaussian at top of file causes circular issues, so using gaussian.Gaussian import within function evaluation